documentation/hacker-guide/source/runtime-manager/access-path.rst

```diff
@@ -0,0 +1,3535 @@
+application). It might even be split across more than one machine or
+more than one process.  In any case, the Debugger Nub is the part that
+knows how to read and write from the runtime environment, set and
+clear breakpoints, and so on. This part of the architecture is
+target-specific.  The access path is able to invoke debugger nubs on
+remote machines by means of a “Debugger Connection” to the remote
+machine. This connection can be thought of as a connection to a server
+process on that remote machine, which has the responsibility of
+invoking the debugger nub, on demand.
+
+The “Spy” is a very lightweight component that is physically part of
+the application. It cooperates closely with the Debugger Nub and
+native debugging API’s (such as ``ptrace`` or :file:`/proc`), which
+provide most of the debugging functionality. At minimum, this part
+preserves a mapping between static handles and objects that are
+subject to relocation by the garbage collector. (The access path
+exposes an interface to this mapping, which we call proxy objects, or
+just proxies.)  Nearly all of the functionality described in this
+document is provided either in the development environment or by the
+Debugger Nub. The one exception to this in the main set of protocols
+is the Remote Object Registration, which is provided by the Spy. Other
+things, such as language-specific remote function call, will also be
+provided in the Spy.  The running application contains Dylan code, and
+may also contain “foreign” code.
+
+The Derived Database contains information about most or all of the
+Dylan code, but will contain no information about the foreign
+code. The Derived Database does not contain enough information to be
+able to look up the name of variable from its address or get a source
+line number from a PC; this information is pinned down during linking,
+and so is stored in the “Debug Info” part of the application.  Since
+applications may contain both Dylan and foreign code, this
+architecture supports debugging of both Dylan and foreign code. The
+access path exposes all of the necessary functionality to make this
+work.
+
+The ACCESS-PATH Module
+======================
+
+The following sections describe the interface that attaches the
+development environment (the “local” side) to an application being
+debugger (the “remote” side).
+
+.. current-module:: access-path
+
+Creating and Attaching Access Paths
+-----------------------------------
+
+The following functions are used to create an access path with a
+particular application running, or to attach a remote process to an
+access path.
+
+.. class:: <access-path>
+   :abstract:
+   :instantiable:
+
+   :superclasses: :class:`<object>`
+
+   :keyword application:
+   :keyword arguments:
+   :keyword process:
+   :keyword core-file:
+   :keyword required application-object: An instance of :class:`<object>`.
+   :keyword symbol-file-locations: An instance of :class:`<sequence>`.
```

Comment:
Names in the `dylan` module should use the `:drm:` role.

Reply:
It would be nice if `environment-reports` would do this automatically.